python/ray/data/dataset.py

```diff
@@ -2720,6 +2726,63 @@ def _experimental_lazy(self) -> "Dataset[T]":
         self._lazy = True
         return self
 
+    @DeveloperAPI
+    def serialize_out_of_band(self) -> bytes:
+        """
+        Serialize the Dataset for out-of-band use, i.e. for use across different Ray
+        clusters. This method serializes the lineage of the Dataset operations. Note
+        that this will drop all computed data, and that everything will be recomputed
```

Comment:
Dumb question: why don't we serialize the computed data (i.e. arrow table, pandas dataframe), is it because of the size?

Reply:
Yes, it's because of the size!